### PR TITLE
Migrated from Users API to OAuth 2.0

### PR DESCRIPTION
**Pull request by Krechals**
From now on, the website is using OAuth 2.0 (authorization framework), instead of Users API.
Preview link: https://topandrei-step-2020.ew.r.appspot.com.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/ServerCommentsServlet.java [hunk 1]
```diff
@@ -49,11 +49,11 @@ private static String evaluateComment(String userComment) {
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     UserService userService = UserServiceFactory.getUserService();
 
-    // Only logged-in users can comment
-    if (!userService.isUserLoggedIn()) {
-      response.sendRedirect("/contact.html?error=login-required");
-      return;
-    }
+    // Only logged-in users can comment. Deprecated!
```
Comment (jeadorf) [inline portfolio/src/main/java/com/google/sps/servlets/ServerCommentsServlet.java]: No need to keep out-commented code around. You can always retrieve it from git revision history.

File: portfolio/src/main/webapp/contact.html [hunk 2]
```diff
@@ -3,19 +3,13 @@
...
-
-    <!-- Bootstrap open-source CSS framework -->
-    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.5.0/css/bootstrap.min.css" integrity="sha384-9aIt2nRpC12Uk9gS9baDl411NQApFmC26EwAOH8WgZl5MYYxFfc+NcPb1dKGj7Sk" crossorigin="anonymous">
-    
-    <script src="https://cdn.jsdelivr.net/npm/sweetalert2@9"></script>
-    <script src="script.js"></script>
-    <script src="comments.js"></script>
-    <link rel="stylesheet" href="style.css">
-    </head>
+    <meta name="google-signin-clinet-id" content="">
```
Comment (jeadorf) [inline portfolio/src/main/webapp/contact.html]: What's the meta tag above? This seems both a typo and redundant.

File: portfolio/src/main/webapp/oauth.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+function signIn(googleUser) {
+  const profile = googleUser.getBasicProfile();
```
Comment (jeadorf) [inline portfolio/src/main/webapp/oauth.js:3]: whitespace, see javascript style guide

**Review by jeadorf — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/oauth.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+
+function signIn(googleUser) {
+  const profile = googleUser.getBasicProfile();
+  const imagurl=profile.getImageUrl();
```
Comment (jeadorf) [inline portfolio/src/main/webapp/oauth.js]: please apply the javascript style guide (see the section on spacing in assignments) also the the rest of the function.

Comment (jeadorf) [inline portfolio/src/main/webapp/oauth.js]: nit: imageUrl instead of imagurl?

**Review by jeadorf — APPROVED** [review]

File: portfolio/src/main/webapp/oauth.js (hunk 4 above)
Comment (Krechals) [reply portfolio/src/main/webapp/oauth.js]: I'm sorry! Previously I didn't see this.

**Pull request merged**